server/src/main/java/org/elasticsearch/search/query/QueryPhaseCollector.java

```diff
@@ -374,11 +373,10 @@ boolean incrementHitCountAndCheckThreshold() {
         }
     };
 
+    @Override
     public void doPostCollection() throws IOException {
-        if (aggsCollector instanceof BucketCollector.BucketCollectorWrapper bucketCollectorWrapper) {
-            bucketCollectorWrapper.bucketCollector().postCollection();
-        } else if (aggsCollector instanceof InternalProfileCollector profileCollector) {
-            profileCollector.doPostCollection();
+        if (aggsCollector instanceof TwoPhaseCollector twoPhaseCollector) {
+            twoPhaseCollector.doPostCollection();
```

Comment:
Should `aggsCollector` always be a `TwoPhaseCollector`? Since `BucketCollectorWrapper` and `InternalProfileCollector` implement that interface?

Reply:
Yes, but trying to exploit that was a headache because some tests are abusing the Collector interface so I am leaving this for a follow up